disaster-recovery-builder/src/main/scala/uk/gov/nationalarchives/builder/Database.scala

```diff
@@ -30,11 +27,13 @@ object Database:
     )
 
     override def write(files: List[OcflFile]): F[Unit] = {
-      val sql = "insert into files (version, id, name, fileId, zref, path, fileName) values (?, ?, ?, ?, ?, ?, ?)"
-      for {
-        logger <- Slf4jLogger.create[F]
-        updateCount <- Update[OcflFile](sql).updateMany(files).transact(xa)
-        _ <- logger.info(s"Created $updateCount rows")
+      val deleteSql = "delete from files where id = ?"
+      val insertSql =
+        "insert into files (version, id, name, fileId, zref, path, fileName, ingestDateTime, sourceId, citation) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+      val deleteAndInsert = for {
+        _ <- Update[UUID](deleteSql).updateMany(files.map(_.id))
+        _ <- Update[OcflFile](insertSql).updateMany(files)
```

Comment:
I initially took it out because the return values in this for comprehension are `ConnectionIO[Int]` which can't be flatMapped with the `F[Unit]` that comes out of the logging calls. 

I've changed it a bit now so it returns `ConnectionIO[(Int, Int)]` with the delete and update count so we can have the message `s"$updateCount rows updated. $deleteCount rows deleted"`